cmd/query/app/builder/storage.go

```diff
@@ -43,22 +42,21 @@ var (
 )
 
 // NewStorageBuilder creates a StorageBuilder based off the flags that have been set
-func NewStorageBuilder(opts ...basicB.Option) (StorageBuilder, error) {
-	flag.Parse()
+func NewStorageBuilder(parsedFlags *flags.Flags, opts ...basicB.Option) (StorageBuilder, error) {
```

Comment:
then another alternative is to again combine the flags/options struct with this logic.